tests/test_field_for_schema.py

```diff
@@ -203,6 +201,25 @@ def test_optional_multiple_types(self):
             ),
         )
 
+    def test_optional_multiple_types_ignoring_union_field_order(self):
+        result = field_for_schema(Optional[Union[int, str]])
```

Comment:
Maybe add a link for context here ?

```suggestion
        # see https://github.com/lovasoa/marshmallow_dataclass/pull/246#issuecomment-1722204048
        result = field_for_schema(Optional[Union[int, str]])
```